File: infer.py
from tinygrad.helpers import getenv, colored, trange
from tinygrad.nn.datasets import mnist
from examples.beautiful_mnist import Model
import time

if __name__ == "__main__":
  X_train, Y_train, X_test, Y_test = mnist()
  
  model = Model()
  res = (model(X_test).argmax(axis=1) == Y_test).mean()*100

  start = time.time()
  list = res.tolist()
  print(f"tolist() took {(time.time() - start):.4f} seconds {list}")

File: infer.ts
import { mnist } from './src/nn/datasets.ts'
import { get_env } from './src/helpers.ts'
import { MNIST } from './beautiful_mnist.ts'

if (import.meta.main) {
  const [X_train, Y_train, X_test, Y_test] = await mnist(undefined, !!get_env('FASHION'))
  const model = new MNIST()
  const res = (model.call(X_test).argmax(1).eq(Y_test)).mean().mul(100)

  const s = performance.now()
  const list = await res.tolist()
  console.log(`Took ${performance.now() - s}, ${list}`)
}
